[UI] Get onboarding code

kuby opened the pull request:
@jkralik Extended web_configuration for the UI:

```json
{
  "domain": "",
  "httpGatewayAddress": "",
  "webOAuthClient": {
    "clientID": "",
    "audience": "",
    "scopes": []
  },
  "deviceOAuthClient": {
    "clientID": "",
    "audience": "",
    "scopes": [],
    "providerName": ""
  }
}
```

kuby committed:
[UI] Get onboarding code

oskarbauer commented on http-gateway/web/src/components/copy-box/index.js:
```diff
@@ -0,0 +1,93 @@
+import { copyToClipboard } from '@/common/utils'
+import { messages as t } from './copy-box-i18n'
+
+class _CopyBox extends PureComponent {
```
how come not a functional component?

kuby replied:
Mixing things up for fun! :) No real reason really, just felt like this could be a Class component.

oskarbauer commented on http-gateway/web/src/containers/things/things-i18n.js:
```diff
@@ -357,4 +361,40 @@ export const messages = defineMessages({
+  },
+  enterDeviceId: {
+    id: 'things.enterDeviceId',
+    defaultMessage: 'Enter device ID',
```
```suggestion
    defaultMessage: 'Enter the device ID',

```

oskarbauer commented on http-gateway/web/src/containers/things/things-i18n.js:
```diff
@@ -357,4 +361,40 @@ export const messages = defineMessages({
+  },
+  provisionNewDevice: {
+    id: 'things.provisionNewDevice',
+    defaultMessage: 'Provision new device',
```
```suggestion
    defaultMessage: 'Provision a new device',

```

oskarbauer commented on http-gateway/web/src/containers/things/things-i18n.js:
```diff
@@ -357,4 +361,40 @@ export const messages = defineMessages({
+  },
+  getCode: {
+    id: 'things.getCode',
+    defaultMessage: 'Get Code',
```
```suggestion
    defaultMessage: 'Get the Code',

```

kuby commented:
@jkralik @ondrejtomcik
- Renamed `auth_config.json` to `web_configuration.json`
- Updated the `web_configuration.json` structure
- Updated the `docs` dashboard section about the config
- Changed `scope` to `scopes` array

jkralik committed:
coap-gateway/http-gateway: onboard device updated

* coap-gateway/http-gateway: allow user to onboard device
via authorization code from UI
